codec/src/main/java/io/netty/handler/codec/compression/AbstractClosableCompressionDecoder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * A skeletal {@link ClosableCompressionDecoder} implementation.
+ */
+public abstract class AbstractClosableCompressionDecoder extends AbstractCompressionDecoder
+        implements ClosableCompressionDecoder {
+
+    private static final InternalLogger logger =
+            InternalLoggerFactory.getInstance(AbstractClosableCompressionDecoder.class);
+
+    protected AbstractClosableCompressionDecoder(CompressionFormat format) {
+        super(format);
+    }
+
+    @Override
+    public abstract boolean isClosed();
+
+    @Override
+    protected void handlerRemoved0(ChannelHandlerContext ctx) throws Exception {
+        if (!isClosed()) {
+            logger.warn("ClosableCompressionDecoder is removed from a pipeline, "
+                    + "but the end of the compressed stream has not been reached.");
```

Comment:
It seems like you already did what I suggested in `AbstractClosableCompressionEncodier.handlerRemoved()`. Perhaps you could do the same here?